include/swift/Driver/ToolChain.h

```diff
@@ -83,6 +86,45 @@ class ToolChain {
     /// arguments.
     const char *getTemporaryFilePath(const llvm::Twine &name,
                                      StringRef suffix = "") const;
+
+    /// Test the number of files to see if an input file list is needed.
```

Comment:
Thinking about it, I think these comments are at the wrong level:
- They should include what Action they're intended to be used for.
- They should *not* include that they're based on the number of files.
- They should not just echo the name of the function, which some of the ones below are doing. Better to have no comment in that case.

Reply:
This is a great catch! Have taken a shot at improvement.